lib/chef/http.rb

```diff
@@ -471,7 +470,11 @@ def retrying_http_errors(url)
     end
 
     def version_retries
-      @version_retries ||= options[:version_class].possible_requests
+      @version_retries ||= if options[:version_class]
+                             options[:version_class].possible_requests
+                           else
+                             0
```

Comment:
not sure this is correct because we could be on a server and client (in the future) that have both dropped support for v0

Reply:
I'm obviously not the expert on this, but this code is not returning the version of the protocol, it's returning the number of times to retry the request:

https://github.com/chef/chef/blob/09cdc96cb25956b3ef053fbe093d06453a7af92b/lib/chef/http.rb#L157

I set it to 0 to indicate "do not retry". Like I said in slack this is not great because you're still getting an error (HTTP 406), just not the one in our code (NoMethodError).